src/compiler/parser.ts

```diff
@@ -594,7 +599,7 @@ namespace ts {
             // If this is a javascript file, proactively see if we can get JSDoc comments for
             // relevant nodes in the file.  We'll use these to provide typing informaion if they're
             // available.
-            if (isJavaScript(fileName)) {
+            if (isSourceFileJavaScript(sourceFile)) {
                 addJSDocComments();

```

Comment:
Alternatively, and perhaps better, we could parse the JS Doc comment as we are parsing the construct it is attached to. In other words, parse the JS Doc comments when we parse the VariableStatement, FunctionDeclaration, or Parameter it is attached to. Any reason it wasn't done that way in the first place?